### PR TITLE
Implemented: usage of auth plugin for single sign on

### PR DESCRIPTION
**Pull request by k2maan**
### Related Issues

Closes #

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/hotwax/job-manager#contribution-guideline)

### CODE REVIEW

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/main.ts [hunk 1]
```diff
@@ -28,13 +28,41 @@ import '@ionic/vue/css/display.css';
...
+import i18n, { translate } from './i18n'
 import store from './store'
 import permissionPlugin from '@/authorization';
 import permissionRules from '@/authorization/Rules';
 import permissionActions from '@/authorization/Actions';
+import { initialise } from '@/adapter'
+import { dxpComponents } from 'dxp-components'
+import { getAndSetUserDetails, getUserTokenAndOms, confirmSessionEnd, logout } from './user-utils';
 
+initialise({
```
[the remark this answers is not in the document]
Comment (k2maan) [reply src/main.ts]: Updated

File: src/App.vue [hunk 2]
```diff
@@ -85,22 +85,23 @@ export default defineComponent({
...
+    //     unauthorised: this.unauthorized,
+    //     responseErrror: () => {
+    //       setTimeout(() => this.dismissLoader(), 100);
+    //     },
+    //     queueTask: (payload: any) => {
+    //       emitter.emit("queueTask", payload);
+    //     }
+    //   }
+    // })
+    // app.use(dxpComponents)
```
[the remark this answers is not in the document]
Comment (k2maan) [reply src/App.vue]: Updated sir

File: src/router/index.ts [hunk 3]
```diff
@@ -25,18 +27,51 @@ declare module 'vue-router' {
...
+const authGuard = async (to: any, from: any, next: any) => {
+  const authStore = useAuthStore()
+  if (!authStore.isAuthenticated) {
+    await loader.present()
+    const token: any = await authStore.authenticate()
+    // redirect if the login fails
+    if (!token?.value?.length) {
+      const redirectUrl = window.location.origin
+      // redirect to launchpad login
+      window.location.href = `http://launchpad.hotwax.io/login?redirectUrl=${redirectUrl}`
```
Comment (adityasharma7) [inline src/router/index.ts:60]: Following URL should be configurable:
`http://launchpad.hotwax.io/login`

File: src/store/modules/user/actions.ts [hunk 4]
```diff
@@ -8,39 +8,45 @@ import { translate } from '@/i18n'
...
+  //       return Promise.reject(new Error(resp.data._ERROR_MESSAGE_));
+  //     }
 
-      const token = resp.data.token;
+  //     const token = resp.data.token;
+  async getAndSetUserDetails({ commit, dispatch, state }, payload) {
+    try {
+      const { token, oms } = payload
+      const baseURL = process.env.VUE_APP_BASE_URL
+      const alias = process.env.VUE_APP_ALIAS ? JSON.parse(process.env.VUE_APP_ALIAS) : {}
```
Comment (adityasharma7) [inline src/store/modules/user/actions.ts:47]: I think this will be part of login plugin

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/user/actions.ts [hunk 5]
```diff
@@ -8,39 +8,45 @@ import { translate } from '@/i18n'
...
+  //       logger.error("error", resp.data._ERROR_MESSAGE_);
+  //       return Promise.reject(new Error(resp.data._ERROR_MESSAGE_));
+  //     }
 
-      const token = resp.data.token;
+  //     const token = resp.data.token;
+  async getAndSetUserDetails({ commit, dispatch, state }, payload) {
+    try {
+      const { token, oms } = payload
+      const baseURL = process.env.VUE_APP_BASE_URL
```
Comment (adityasharma7) [inline src/store/modules/user/actions.ts:46]: This will be part of login plugin only

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/user/actions.ts (hunk 5 above)
Comment (k2maan) [reply src/store/modules/user/actions.ts]: Removed sir, this is handled in the Login plugin in the Launchpad app.

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**